app/src/main/java/ch/epfl/qedit/frontendBackendInterface/Request.java

```diff
@@ -0,0 +1,7 @@
+package ch.epfl.qedit.frontendBackendInterface;
+
+public abstract class Request extends FallibleEntity {
+    public Request() {
```

Comment:
Why do you always create an abstract class before creating a class ?

Reply:
Some of them will be turned into interface like @brunnerant suggested ... But there will be more types of requests than just the request of login. So the abstract class or the interface allows the group common code/properties for all request. Keep in mind, this is just the first version of a structure of classes and interfaces, it will be extended in the future.

Reply:
So what's going to happen ? We want a request for a name and you will create a RequestName ? You want a request for a quiz you create a class RequestQuiz ? You want a request for a banana you create a class RequestBanana ? It will become unmanageable and a big mess.

In Firebase side it is quite simple to login, usually just a function for example with password and email just, firebase.signinwithemailpassword(email, password). You seem to add a lot of layer of complexity from our side.

Reply:
Yes, I already implemented the RequestBanana.

Reply:
Until now I implemented more or less what we have discussed last Friday. Maye it would be better to simply add methods to Backend or Server instead of creating the different classes of Requests and Responses.

Reply:
I think we will have to discuss with the other more in detail about that.

Reply:
what i propose for this pull request is to wait until the sprint review, and we will decide the correct architecture friday morning. We will get the work redistributed then because we will need to put the back end front end and interface working together for the next sprint